hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/security/acl/RequestContext.java

```diff
@@ -43,20 +43,23 @@
    */
   private final boolean recursiveAccessCheck;
 
+  private final OzonePrefixPath ozonePrefixPath;
+
   @SuppressWarnings("parameternumber")
   public RequestContext(String host, InetAddress ip,
       UserGroupInformation clientUgi, String serviceId,
       ACLIdentityType aclType, ACLType aclRights,
       String ownerName) {
     this(host, ip, clientUgi, serviceId, aclType, aclRights, ownerName,
-            false);
+            false, null);
   }
 
   @SuppressWarnings("parameternumber")
   public RequestContext(String host, InetAddress ip,
       UserGroupInformation clientUgi, String serviceId,
       ACLIdentityType aclType, ACLType aclRights,
-      String ownerName, boolean recursiveAccessCheck) {
+      String ownerName, boolean recursiveAccessCheck,
+      OzonePrefixPath ozonePrefixPath) {
```

Comment:
Should we put the OzonePrefixPath in the OzoneObject?

Reply:
Thanks @xiaoyuyao for the comment. Agreed.

I hope you are thinking that, OzoneObject has reference to `volume, bucket and keyname` attributes, so we can also add `ozonePrefixPath` to OzoneObject to make it more sense. Am I correct?

Is that `recursiveAccessCheck` flag in RequestContext work instead of adding a new flag to indicates the check for prefixPath? 

Now, `recursiveAccessCheck` flag will be true for any directory and false otherwise. I'm setting the value in OMKeyRenameRequestV1 and OMKeyDeleteRequestV1.

Reply:
@xiaoyuyao , I have updated the patch by addressing your comment. Thanks for the reviews!